analyzer/src/main/kotlin/Analyzer.kt

```diff
@@ -74,14 +74,16 @@ class Analyzer(private val config: AnalyzerConfiguration, private val labels: Ma
             "Using the following configuration settings:\n${yamlMapper.writeValueAsString(repositoryConfiguration)}"
         }
 
+        val distinctPackageManagers = packageManagers.distinct()
```

Comment:
I'd also prefer failing here, e.g. use require instead of distinct. (Same in several other places below)

Reply:
Here, I believe any duplicates do no harm, and the deduplication is basically just a performance optimization to not find definition files with the same package manager twice.

Why do you believe it should be an error if the user says "find files with Gradle, Gradle and NPM" instead of just saying "find files with Gradle and NPM"? I.e. what's the risk?

Reply:
In my view tolerating duplicate values as command line option is fine. However, I believe that should be handled only in the `AnalyzerCommand`, not all over the place. 

In other places (assuming that we all agree on this change which puts the factories by name on the map) I believe the assumption should be made that pkg mgrs are always `disctinctBy` their name, to avoid the IDs in the ORT file becoming ambigeous.

Reply:
In my view tolerating duplicate values as command line option is fine. However, I believe that should be handled only in the `AnalyzerCommand`, not all over the place. 

In other places (assuming that we all agree on this change which puts the factories by name on the map) I believe the assumption should be made that pkg mgrs are always `disctinctBy` their name, to avoid the IDs in the ORT file becoming ambigeous.

Reply:
> However, I believe that should be handled only in the `AnalyzerCommand`, not all over the place.

More concretely, are you suggesting to keep the `packageManagers` parameter type at a `Set` instead of a `Collection`? I thought I'm doing you a favor here, as it's usually you who advocates for functions to take more generic collections.

Reply:
> More concretely, are you suggesting to keep the `packageManagers` parameter type at a `Set` instead of a `Collection`?

No. My suggestion was to use a `require` block as I wrote above.

> I thought I'm doing you a favor here, as it's usually you who advocates for functions to take more generic collections.

If there is no need for a parameter type to be "more specific" I believe that it's better to stick with the more general type, to allow for calling the function more easily (without prior conversion). This is what I usually try to adhere to when choosing the parameter type. Not sure if this is what you're talking about.

Reply:
> No. My suggestion was to use a `require` block as I wrote above.

That doesn't make sense to me then, sorry. I don't see a point in failing hard here if we either could

- enforce "intended" use by keeping the function parameter type at a `Set` instead of a `Collection`, or
- "recover" from "unintended" use by simply deduplicating the collection.

Reply:
> enforce "intended" use by keeping the function parameter type at a Set instead of a Collection, or

That doesn't make sense to me because making the type a `Set` makes the factories distinct by instance, while as per the introduce map they should be distinct by their name, sorry.

> "recover" from "unintended" use by simply deduplicating the collection.

I'm not against deduplicating at all. I doesn't make sense to me to do this in multiple places rather than just in the CLI, sorry again.

Anyhow, as [1] has been resolved I won't block this and resolve this discussion as well.

[1] https://github.com/oss-review-toolkit/ort/pull/6015#discussion_r1010177969

Reply:
> I doesn't make sense to me to do this in multiple places rather than just in the CLI, sorry again.

Please keep in mind that the `findManagedFiles` functions are part of the public API which might be used programmatically e.g. in the ORT server context.